README.md

```diff
@@ -2,3 +2,209 @@
 
 ![](https://github.com/mikanikos/Fork-Accountability/workflows/Build%20and%20Tests/badge.svg)
 [![codecov](https://codecov.io/gh/mikanikos/Fork-Accountability/branch/master/graph/badge.svg)](https://codecov.io/gh/mikanikos/Fork-Accountability)
+
+The scope of this project is to implement a simple PoC of a Fork Accountability algorithm in the Tendermint Consensus protocol.
+
+The repository includes simple libraries and modules that allow to run experiments and benchmark the algorithm implementation in a minimal test environment:
+
+- a simple connection library to provide an high-level API for establishing connections and communicating among processes based on a client-server architecture; 
+
+- an accountability algorithm implementation based on the theoretical specifications given by the research scientists of the DCL lab at EPFL and Interchain Foundation;  
+
+- monitor implementation that represents the independent verification entity used to run the accountability algorithm;  
+
+- validator implementation that represents the processes that participate in the Tendermint Consensus protocol; 
+
+- sample test scripts to easily run experiments and test both the accountability algorithm and the interaction between monitor and validators.
+
+## Context and overview
+Documentation describing the context and the theoretical background can be found in the [docs](docs) folder.   
+
+## Architecture
+
+The project is organized in several files and packages in order to guarantee modularity and extendability and maintain at the same time a clear and simple structure.
+
+The monitor (monitor package) is the entity responsible to run the accountability algorithm. It takes as input parameters:
```

Comment:
I couldn't find the monitor package, maybe I'm missing something -- did you mean `cmd/monitor/`? Can you add a precise reference to it pls?

Reply:
Done